docs/extension-development/check-if-woo-is-active.md

```diff
@@ -4,28 +4,45 @@ menu_title: Check if WooCommerce is active
 tags: how-to
 ---
 
-You can wrap your plugin in a check to see if WooCommerce is installed:
+When developing for WooCommerce, it's important to check that WooCommerce is installed and active before your own code runs. This ensures no errors occur from missing WooCommerce functions or classes.
+
+There are a few methods to achieve this. The most reliable approach is to initiate your code on `plugins_loaded`. This means you can be sure that WooCommerce has already initiated, by checking if one of its core classes exists.
 
 ```php
-// Test to see if WooCommerce is active (including network activated).
+add_action( 'plugins_loaded', 'prefix_plugins_loaded' );
 
-$plugin_path = trailingslashit( WP_PLUGIN_DIR ) . 'woocommerce/woocommerce.php';
+function prefix_plugins_loaded() {
+	// Check if the "WooCommerce" class exists. If not, exit early.
+	if ( ! class_exists( 'WooCommerce' ) ) {
+		return;
+	}
 
-if (
+	// Custom code here. WooCommerce is active and initiated...
+}
+```
 
-in_array( $plugin_path, wp_get_active_and_valid_plugins() )
+Another route you could take is to initiate your code on the `woocommerce_init` hook. We know this only runs when WooCommerce is active and initiated.
 
-|| in_array( $plugin_path, wp_get_active_network_plugins() )
+```php
+add_action( 'woocommerce_init', 'prefix_woocommerce_init' );
 
-) {
+function prefix_woocommerce_init() {
+	// Custom code here. WooCommerce is active and initiated...
+}
```

Comment:
There is also `before_woocommerce_init` which could be used (and execution is on the init hook but _right before_ all WooCommerce initiatlizes).